query-engine/core/src/query_graph_builder/write/utils.rs

```diff
@@ -172,20 +173,14 @@ pub fn insert_1to1_idempotent_connect_checks(
     let child_model = parent_relation_field.related_model();
     let child_model_identifier = child_model.primary_identifier();
 
-    let diff_node = graph.create_node(Node::Computation(Computation::empty_diff()));
+    let diff_node = graph.create_node(Node::Computation(Computation::empty_diff_left_to_right()));
 
     graph.create_edge(
         read_new_child_node,
         &diff_node,
-        QueryGraphDependency::ProjectedDataDependency(
+        QueryGraphDependency::ProjectedDataSinkDependency(
             child_model_identifier.clone(),
-            Box::new(move |mut diff_node, child_ids| {
-                if let Node::Computation(Computation::Diff(ref mut diff)) = diff_node {
-                    diff.right = child_ids.into_iter().collect();
-                }
-
-                Ok(diff_node)
-            }),
+            DataSink::AllRows(&LeftSideDiffInput),
```

Comment:
Note that `diff.right` was changed to `LeftSideDiffInput` here. We could've also kept this as `RightSideDiffInput` and change the node type to `DiffRightToLeft`, this order just felt more natural.